components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/PolicyDAOImpl.java

```diff
@@ -69,29 +69,39 @@ public Policy getPolicy(String policyLevel, String policyName) throws APIMgtDAOE
                 return getApplicationPolicy(policyName);
             } else if (APIMgtConstants.ThrottlePolicyConstants.SUBSCRIPTION_LEVEL.equals(policyLevel)) {
                 return getSubscriptionPolicy(policyName);
+            } else {
+                throw new APIMgtDAOException("Couldn't find the policy for name: " + policyName + ", level: " +
+                        policyLevel);
             }
         } catch (SQLException e) {
             throw new APIMgtDAOException("Couldn't find " + policyName + ' ' + policyLevel + "Policy", e);
         }
-        return null;
     }
 
+    /**
+     *@see PolicyDAO#getPolicyByUuid(String, String)
+     */
     @Override
-    public Policy getPolicyByUuid(String uuid, String policyLevel) throws APIMgtDAOException {
+    public Policy getPolicyByUuid(String policyLevel, String uuid) throws APIMgtDAOException {
```

Comment:
I understand what you mean by not having to add a new function, this makes the life of the API developer easier. But it does not mean that the API user has it easy. Priority should go to the API user to reduce the chance of them making a mistake.

Actually when you add a new function to handle a new policy type the existing API does not change, it just gets additional methods. So existing code using your current API will not get affected. Code that wants to use the new policy functions needs to only call the new methods you added